File: gtranslate.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <curl/curl.h>

#define UA "Mozilla/5.0"

#define URL "http://translate.googleapis.com/translate_a/single?client=gtx&sl=auto&tl=ru&dt=t&q=\""

size_t write_data(char *data, size_t size, size_t nmemb);

int main(int argc, char *argv[])
{    
    if (argc != 2) {
        printf("Usage: %s \"text\"\n", argv[0]);
        return 1;
    }    

    CURL *curl = curl_easy_init();    
    if (!curl) {
        fprintf(stderr, "curl error");    
        return 1;
    }

    CURLcode res;
    char *rdata, *urldata;
    int len;
    
    len = strlen(argv[1]);
    rdata = curl_easy_escape(curl, argv[1], len);
    len += strlen(rdata) + 512;
    urldata = (char *)calloc(len, sizeof(char));
    if (urldata == NULL) {
        fprintf(stderr, "memory allocation error");
        curl_easy_cleanup(curl);
        return 1;
    }
    strcpy(urldata, URL);
    strcat(urldata, rdata);
    strcat(urldata, "\"");
    curl_easy_setopt(curl, CURLOPT_URL, urldata);
    curl_easy_setopt(curl, CURLOPT_USERAGENT, UA);
    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_data);
    res = curl_easy_perform(curl);
    if (res != CURLE_OK)
        fprintf(stderr, "Response error");
    curl_easy_cleanup(curl);
    free(urldata);

    return 0;
}


size_t write_data(char *data, size_t size, size_t nmemb)
{
    char *p = data + 4;
    char *start, *t;
    
    if (*p == '\\' && *(p + 1) == '\"')
        p += 2;

    start = p;
    while (*p) {
        if ((t = strstr(p, "\",\"")) != NULL) {
            
            if (*(t - 2) == '\\' && *(t - 1) == '\"')
                p = t - 2;
            else
                p = t;
            printf("%.*s", p - start, start);
        }
        
        if ((t = strstr(p, "],[\"")) != NULL)
            start = p = t + 4;
        
        ++p;
    }
        
    putchar('\n');
    
    return  size * nmemb;
}

File: gtranslate.go
package main

import (
	"fmt"
	"io/ioutil"
	"net/http"
	"net/url"
	"os"
	"strings"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Fprintf(os.Stderr, "Usage: %s \"text\"\n", os.Args[0])
		os.Exit(1)
	}
	echeck := func(err error) {
		if err != nil {
			fmt.Fprintln(os.Stderr, err)
			os.Exit(1)
		}
	}
	Url := "http://translate.googleapis.com/translate_a/single?client=gtx&sl=auto&tl=ru&dt=t&"
	Ua := "Mozilla/5.0"

	client := &http.Client{}
	s := make(url.Values)
	s.Add("q", strings.TrimSpace(os.Args[1]))
	Url = Url + s.Encode()
	req, err := http.NewRequest("GET", Url, nil)
	echeck(err)
	req.Header.Set("User-Agent", Ua)
	resp, err := client.Do(req)
	echeck(err)
	data, err := ioutil.ReadAll(resp.Body)
	resp.Body.Close()
	echeck(err)

	str := strings.ReplaceAll(string(data[4:]), "\n", "")
	length := len(str)
	for i := 0; i < length; i++ {
		if n := strings.Index(str[i:], "\",\""); n != -1 {
			if i == 0 {
				fmt.Fprintf(os.Stdout, "%s", str[i:i+n])
			}
			if m := strings.Index(str[i:i+n], "],["); m != -1 {
				for _, t := range str[i+m+4 : i+n] {
					if t >= 1000 {
						fmt.Fprintf(os.Stdout, "%s", str[i+m+4:i+n])
						break
					}
				}
			}
			i += n + 3
		} else {
			break
		}
	}
	fmt.Println()
}
